### DISCUSSION TITLE
What is exactly the cost of a createBuffer at each frame ?

### DISCUSSION THREAD

**Discussion by ra1jin** [gpuweb/gpuweb#3458] [Q&A]
Hello, in the context of a graphics manager we want to know what is the best way to handle a large vertex buffer.

**WAY 1:** 
At each frame destroy buffer -> create a new one -> write data with javascript float32 copy.

**WAY 2:**
Keep the same buffer and make changes with a range system.
This solution feel more efficient but add a lot of complexity so if the first way is not a problem we go with the 1.

**WAY 3:**
Keep a buffer by drawable and destroy when object is deleted.
This solution need to switch buffer inside graphics draw loop that is apparently a slow operation.

**More general ask:**
What is the difference between two ways ?
What is exactly the cost of destroying a buffer and create a completely new at each frame ?

**Comment by munrocket** [comment]
Looks like this issue related to memory allocation and garbage collection in JS engines. They are different in different browsers: V8, JavaScriptCore, SpiderMonkey. Way2 will not have these problems, but there will be another one - complexity. There is no single solution for all projects, you can measure performance by yourself and decide how bad it is. You probably will see only difference in memory allocation. Garbage collection is happening not immediately, you should catch the moment when it is happening.
